File: Leetcode/PY/Leetcode318-最大单词长度乘积.py
class Solution:
    def maxProduct(self, words: List[str]) -> int:
        n = len(words)
        bitmask = [0] * n
        for i in range(n):
            for ch in words[i]:
                t = ord(ch) - ord('a')
                bitmask[i] |= 1 << t
        res = 0
        for i in range(n):
            for j in range(i + 1,n):
                if bitmask[i] & bitmask[j] == 0:
                    res = max(res,len(words[i]) * len(words[j]))
        return res

File: Leetcode/Rust/Leetcode318-最大单词长度乘积.rs
impl Solution {
    pub fn max_product(words: Vec<String>) -> i32 {
        let mut bitmask = vec![0;words.len()];
        for i in 0..words.len(){
            for ch in words[i].as_bytes(){
                let t = ch - b'a';
                bitmask[i] |= 1 << t;
            }
        }
        let mut res = 0;
        for i in 0..words.len(){
            for j in (i + 1)..words.len(){
                if bitmask[i] & bitmask[j] == 0{
                    res = res.max((words[i].len() * words[j].len()) as i32);
                }
            }
        }
        return res;
    }
}